transport/grpc/dial.go

```diff
@@ -135,6 +135,25 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 		transportCreds = credentials.NewTLS(&tls.Config{
 			GetClientCertificate: clientCertSource,
 		})
+		// try mtls with s2a when all the followings are true:
```

Comment:
so this is quite a bit of code to add and some duplicated code for the grpc & http implementations - can you encapsulate some of the logic into helper functions files? Also, as far as implementation goes, is there anyway to integrate s2a as a "clientCertSource" using the existing interface so it can be one of the options on the fallback path? (Both ECP enterprise-cert-proxy and secure-connect are on that code-path)

Reply:
Thanks Andy, I did some refactoring to move the code into separate files/functions, and it's cleaner now.

For the other question, I also thought about it initially, the issue is that S2A also implements other parts of the tls.Config struct, e.g. `VerifyPeerCertificate` method, so certificate source alone is not sufficient.
https://github.com/google/s2a-go/blob/36dafe55c3480837e3470a80970753cc63e0f3f5/internal/v2/tlsconfigstore/tlsconfigstore.go#L112